demes/demes.py

```diff
@@ -1997,34 +1997,39 @@ def in_generations(self) -> Graph:
         graph.generation_time = 1
         return graph
 
-    def rename_demes(self, names: MutableMapping[str, str]) -> Graph:
+    def rename_demes(self, names: Mapping[str, str]) -> Graph:
         """
-        Rename demes according to a dictionary.
+        Rename demes according to a dictionary that may contain a partial set of demes.
 
         :param dict names:
             A dictionary with deme names and new names.
         :return:
             A demographic model with renamed demes.
         :rtype: Graph
         """
+        if not isinstance(names, Mapping):
+            raise TypeError("names is not a dictionary")
         graph = copy.deepcopy(self)
-        if not isinstance(names, MutableMapping):
-            raise ValueError("names is not a mapping!")
         for deme in graph.demes:
-            if deme.name not in names:
-                names[deme.name] = deme.name
-        for deme in graph.demes:
-            deme.name = names[deme.name]
-            deme.ancestors = [names[ancestor] for ancestor in deme.ancestors]
+            if deme.name in names:
+                deme.name = names[deme.name]
+                deme.ancestors = [
+                    names[ancestor] if ancestor in names else ancestor
+                    for ancestor in deme.ancestors
+                ]
```

Comment:
I think this misses the case where `deme.name` is not in `names`, but the deme has ancestors that must be renamed.

Reply:
Good spot! I think that this should now be fixed in the most recent commit as it passes all of the tests now.